test/io/test_octree_compression.cpp

```diff
@@ -0,0 +1,209 @@
+        cloud->push_back(new_point);
+      }
+//      std::cout << "Run: " << total_runs << " compression profile:" << compression_profile << " point_count: " << point_count;
+      std::stringstream compressed_data;
+      try
+      { // decodePointCloud() throws exceptions on errors
+        pointcloud_encoder->encodePointCloud(cloud, compressed_data);
+        pointcloud_decoder->decodePointCloud(compressed_data, cloud_out);
+        EXPECT_GT((int)cloud_out->width, 0) << "decoded PointCloud width <= 0";
+        EXPECT_GT((int)cloud_out->height, 0) << " decoded PointCloud height <= 0 ";
+      }
+      catch (std::exception& e)
+      {
+        std::cout << e.what() << std::endl;
+      }
+    } // runs
+  } // compression profiles
+} // TEST
+
+TEST(PCL, OctreeDeCompressionFile)
+{
+  pcl::PointCloud<pcl::PointXYZRGB>::Ptr input_cloud_ptr (new pcl::PointCloud<pcl::PointXYZRGB>);
+    
+  // load point cloud from file, when present
+  if (pcd_file == NULL) return;
+    int rv = pcl::io::loadPCDFile(pcd_file, *input_cloud_ptr);
+    float voxel_sizes[] = { 0.1, 0.01 };
+    
+    EXPECT_EQ(rv, 0) << " loadPCDFile " << pcd_file;
+    EXPECT_GT((int) input_cloud_ptr->width , 0) << "invalid point cloud width from " << pcd_file;
+    EXPECT_GT((int) input_cloud_ptr->height, 0) << "invalid point cloud heigth from " << pcd_file;
+
+    // iterate over compression profiles
+    for (int compression_profile = pcl::io::LOW_RES_ONLINE_COMPRESSION_WITHOUT_COLOR;
+         compression_profile != pcl::io::COMPRESSION_PROFILE_COUNT; ++compression_profile) {
+    // instantiate point cloud compression encoder/decoder
+    pcl::io::OctreePointCloudCompression<pcl::PointXYZRGB>* PointCloudEncoder = new pcl::io::OctreePointCloudCompression<pcl::PointXYZRGB>((pcl::io::compression_Profiles_e) compression_profile, false);
+    pcl::io::OctreePointCloudCompression<pcl::PointXYZRGB>* PointCloudDecoder = new pcl::io::OctreePointCloudCompression<pcl::PointXYZRGB>();
+ 
+    // iterate over various voxel sizes
+    for (int i = 0; i < sizeof(voxel_sizes)/sizeof(voxel_sizes[0]); i++) {
+      pcl::octree::OctreePointCloud<pcl::PointXYZRGB> octree(voxel_sizes[i]);
+      pcl::PointCloud<pcl::PointXYZRGB>::Ptr octree_out(new pcl::PointCloud<pcl::PointXYZRGB>());
+      octree.setInputCloud((*input_cloud_ptr).makeShared());
+      octree.addPointsFromInputCloud();
+            
+      std::stringstream compressedData;
+      PointCloudEncoder->encodePointCloud(octree.getInputCloud(), compressedData);
+      PointCloudDecoder->decodePointCloud(compressedData, octree_out);
+      EXPECT_GT((int)octree_out->width, 0) << "decoded PointCloud width <= 0";
+      EXPECT_GT((int)octree_out->height, 0) << " decoded PointCloud height <= 0 ";
+      total_runs++;
+    }
+    delete PointCloudDecoder;
+    delete PointCloudEncoder;
+  }
+}
+
+/* ---[ */
+int
+main (int argc, char** argv)
+{
+  testing::InitGoogleTest (&argc, argv);
+  if (argc > 1) {
+    pcd_file = argv[1];
+  }

```

Comment:
This test is wrong. If ```argc``` is less than 2 you should abort the test, otherwise your OctreeDeCompressionFile will explode. The usual behavior is to print a message indicating which file should be supplied to the input. Look at [this example](https://github.com/PointCloudLibrary/pcl/blob/master/test/io/test_ply_mesh_io.cpp#L286-L290).